sstables/sstable_set.cc

```diff
@@ -299,14 +299,20 @@ partitioned_sstable_set::partitioned_sstable_set(schema_ptr schema, bool use_lev
         , _use_level_metadata(use_level_metadata) {
 }
 
+static std::unordered_map<run_id, shared_sstable_run> clone_runs(const std::unordered_map<run_id, shared_sstable_run>& runs) {
+    return boost::copy_range<std::unordered_map<run_id, shared_sstable_run>>(runs | boost::adaptors::transformed([] (auto& p) {
+        return std::make_pair(p.first, make_lw_shared<sstable_run>(*p.second));
```

Comment:
This change looks good.
I just wonder if we really need the runs to be kept as shared pointers.

Reply:
It allows compaction manager to invoke ICS without having to copy all runs (sstable_set returns runs as a set of `lw_shared_ptr<const sstable_run>`). The copy will have to happen anyway when cloning the set. It can also happen ICS retrieves runs, but finds no compaction job, so it would be a waste to copy runs unnecessarily.